src/script/common/c_content.cpp

```diff
@@ -1447,9 +1447,9 @@ ToolCapabilities read_tool_capabilities(
 		int table_groupcaps = lua_gettop(L);
 		lua_pushnil(L);
 		while(lua_next(L, table_groupcaps) != 0){
-			// key at index -2 and value at index -1
-			std::string groupname = luaL_checkstring(L, -2);
 			if(lua_istable(L, -1)){
+				// key at index -2 and value at index -1
+				std::string groupname = luaL_checkstring(L, -2);
```

Comment:
This contains a check. Previous version was better.

Reply:
@SmallJoker, fix it https://github.com/minetest/minetest/pull/12704/commits/7b18f1da0cf9bc70b334d51fc03ca93000e70811